test/parallel/test-events-on-async-iterator.js

```diff
@@ -113,39 +113,6 @@ async function throwInLoop() {
   assert.strictEqual(ee.listenerCount('error'), 0);
 }
 
-async function next() {
```

Comment:
why these tests deleted?

Reply:
I don't quite think they are valid. Also they don't work with an async generator implementation. Please do have a look at it. But I can't find a way to make that test pass with an async generator and assuming async generators are "correct" then the test is "incorrect".

Reply:
> They don't work with an async generator implementation

What is the output with the async iterator implementation? 

These tests specifically check that if the user asked for two events and then closed the iterator they still get those two events and not one event and then the iterator closing. That behavior is quite explicit - though we can talk about changing it (it'd be a breaking change though I doubt a lot of users are relying on this behavior)

Reply:
So the problem here is the `return()` which doesn't actually guarantee that `iterable.next()´ will ever resolve since the async generator will never reach the yield statement while waiting for the promise to resolve. As far as I know there is not way to do this in an async generator. It's usually not a problem for `for await` loops so I'm not quite sure what exactly the spec or users would expect out of this use case?

Reply:
Discussion starting roughly here: https://github.com/nodejs/node/pull/27994/files#diff-330bdd22a188135540523f69deb4f9563e03b00a913cd369e1ee84d899f178a9

Reply:
Can't see it from there?

Reply:
I don't know about the specifics of this but it seems weird to me that we want to achieve a behaviour which is not achievable with an async generator. Maybe I'm a bit too hung up on assuming async generator is the "correct" way.

Reply:
> I don't know about the specifics of this but it seems weird to me that we want to achieve a behaviour which is not achievable with an async generator. 

Not dropping these events is why the implementation didn't use an async generator initially - it's OK to change that if you think the simplification is worth losing that API guarantee.

It's true users who `for... await` will `next()` one at a time but implementations (think `from`) can ask for multiple values at once and this changes the behavior of the last chunks in the stream.

Reply:
> if you think the simplification is worth losing that API guarantee.

I think so. But I'm not sure if my opinion is sufficient here. I don't understand why we wanted this API guarantee to begin with.